cmd/load-runner/helm.go

```diff
@@ -67,11 +87,75 @@ func (h *helmClient) Uninstall(config *parser.LoadTestConfig) error {
 }
 
 func processHelmEnvVars(args []string, envVars map[string]string) []string {
+	args = calculateLoadParameters(args, envVars)
 	for key, value := range envVars {
 		if strings.Contains(value, ",") {
 			value = strings.ReplaceAll(value, ",", "\\,")
 		}
 		args = append(args, "--set", fmt.Sprintf("deployment.env.%s=%s", key, value))
 	}
+
+	return args
+}
+
+func calculateLoadParameters(args []string, envVars map[string]string) []string {
+	resourceCalculation := envVars["RESOURCE_CALCULATION"]
+
+	// Handle auto calculation
+	if resourceCalculation == "auto" {
+		maxEventsPerSecond, err := strconv.Atoi(envVars["MAX_EVENTS_PER_SECOND"])
+		if err != nil {
+			log.Fatalf("Failed to convert MAX_EVENTS_PER_SECOND to int: %v", err)
+		}
+		resourceMultiplier := maxEventsPerSecond/5000 + 1
+		envVars["CONCURRENCY"] = strconv.Itoa(resourceMultiplier * 2000)
+		envVars["MESSAGE_GENERATORS"] = strconv.Itoa(resourceMultiplier * 500)
+		args = append(args, "--set", fmt.Sprintf("deployment.resources.cpuRequests=%d", resourceMultiplier),
+			"--set", fmt.Sprintf("deployment.resources.cpuLimits=%d", resourceMultiplier),
+			"--set", fmt.Sprintf("deployment.resources.memoryRequests=%dGi", resourceMultiplier*2),
+			"--set", fmt.Sprintf("deployment.resources.memoryLimits=%dGi", resourceMultiplier*2),
+		)
+		return args
+	}
+
+	// Handle overprovision calculation
+	if strings.HasPrefix(resourceCalculation, "overprovision,") {
+		parts := strings.Split(resourceCalculation, ",")
+		if len(parts) != 2 {
+			log.Fatalf("Invalid RESOURCE_CALCULATION format for overprovision. Expected 'overprovision,<percentage>'")
+		}
+
+		overprovisionPercent, err := strconv.Atoi(parts[1])
+		if err != nil {
+			log.Fatalf("Failed to convert overprovision percentage to int: %v", err)
+		}
+
+		if overprovisionPercent < 0 || overprovisionPercent > 100 {
+			log.Fatalf("Overprovision percentage must be between 0 and 100, got: %d", overprovisionPercent)
+		}
+
+		maxEventsPerSecond, err := strconv.Atoi(envVars["MAX_EVENTS_PER_SECOND"])
+		if err != nil {
+			log.Fatalf("Failed to convert MAX_EVENTS_PER_SECOND to int: %v", err)
+		}
+
+		// Calculate base resource multiplier
+		baseMultiplier := maxEventsPerSecond/5000 + 1
+
+		// Apply overprovisioning
+		overprovisionFactor := 1.0 + float64(overprovisionPercent)/100.0
+		resourceMultiplier := math.Round(float64(baseMultiplier)*overprovisionFactor*100) / 100
+
+		envVars["CONCURRENCY"] = strconv.Itoa(int(resourceMultiplier * 2000))
+		envVars["MESSAGE_GENERATORS"] = strconv.Itoa(int(resourceMultiplier * 500))
+
+		args = append(args, "--set", fmt.Sprintf("deployment.resources.cpuRequests=%v", resourceMultiplier),
+			"--set", fmt.Sprintf("deployment.resources.cpuLimits=%v", resourceMultiplier),
+			"--set", fmt.Sprintf("deployment.resources.memoryRequests=%vGi", math.Round(resourceMultiplier*2)),
+			"--set", fmt.Sprintf("deployment.resources.memoryLimits=%vGi", math.Round(resourceMultiplier*2)),
+		)
+		return args
```

Comment:
Can we do all this with a function and use it for both `auto` and `overprovision`? With `overprovision` it will just add the overprovision percentage to the same math done for `auto` and then return the same `args` with the same multiplier also for limits. Wdyt?

Reply:
Very good catch! I have unified the implementation. Let me know if you see any further chance for improvement.